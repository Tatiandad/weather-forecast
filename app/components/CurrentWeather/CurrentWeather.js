import React, { Component } from 'react';

class CurrentWeather extends Component {
  render () {
    const { city, country, currentWeather } = this.props;

    return (
      <div className='today-container'>
        <h2>{`Current weather in ${city}, ${country}`}</h2>
        <div className='today-weather'>
          <h3>
            <img 
              src={`https://openweathermap.org/img/w/${currentWeather.weather[0].icon}.png`} 
              alt={currentWeather.weather[0].description}
            />
            {Math.round(currentWeather.main.temp)}&deg;{this.props.unit}
          </h3>
          <p>{currentWeather.weather[0].description}</p>
        </div>
      </div>
    )
  }
}

export default CurrentWeather;