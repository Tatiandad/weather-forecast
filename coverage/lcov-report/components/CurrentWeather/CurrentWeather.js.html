<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/CurrentWeather/CurrentWeather.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/CurrentWeather</a> CurrentWeather.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
&nbsp;
class CurrentWeather extends Component {
  render () {
    const { city, country, currentWeather } = this.props;
&nbsp;
    return (
      &lt;div className='today-container'&gt;
        &lt;h2&gt;{`Current weather in ${city}, ${country}`}&lt;/h2&gt;
        &lt;div className='today-weather'&gt;
          &lt;h3&gt;
            &lt;img 
              src={`http://openweathermap.org/img/w/${currentWeather.weather[0].icon}.png`} 
              alt={currentWeather.weather[0].description}
            /&gt;
            {Math.round(currentWeather.main.temp)}&amp;deg;{this.props.unit}
          &lt;/h3&gt;
          &lt;p&gt;{currentWeather.weather[0].description}&lt;/p&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default CurrentWeather;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 17 2018 17:59:23 GMT-0500 (Eastern Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
