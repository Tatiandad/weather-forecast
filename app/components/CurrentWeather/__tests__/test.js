import React, { Component } from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import "babel-polyfill";
import { shallow } from 'enzyme';
import CurrentWeather from '../CurrentWeather';
import weather from '../../../utils/dummy-weather';

configure({ adapter: new Adapter() });

const dummyProps = {
  city: 'London',
  country: 'GB',
  unit: 'F',
  currentWeather: {
    main: weather.main,
    weather: weather.weather,
    wind: weather.wind
  }
};

let shallowWrapper;

beforeEach(() => {
  shallowWrapper = shallow(<CurrentWeather {...dummyProps}/>);
});

describe('<CurrentWeather />', () => {
  it('renders <CurrentWeather /> component', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('renders elements of <Day /> correctly', () => {
    const wrapperDiv = shallowWrapper.find('.today-container');
    const header = shallowWrapper.find('.today-weather p');
    expect(wrapperDiv.length).toBe(1);
    expect(header.text()).toEqual(dummyProps.currentWeather.weather[0].description);
  });
});