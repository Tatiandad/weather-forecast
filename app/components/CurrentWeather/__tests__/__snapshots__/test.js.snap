// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentWeather /> renders <CurrentWeather /> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentWeather
    city="London"
    country="GB"
    currentWeather={
      Object {
        "main": Object {
          "humidity": 91,
          "pressure": 1010,
          "temp": 39.78,
          "temp_max": 41,
          "temp_min": 37.94,
        },
        "weather": Array [
          Object {
            "description": "moderate rain",
            "icon": "10n",
            "id": 501,
            "main": "Rain",
          },
          Object {
            "description": "mist",
            "icon": "50n",
            "id": 701,
            "main": "Mist",
          },
        ],
        "wind": Object {
          "deg": 20,
          "gust": 7.7,
          "speed": 10.29,
        },
      }
    }
    unit="F"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Current weather in London, GB
        </h2>,
        <div
          className="today-weather"
        >
          <h3>
            <img
              alt="moderate rain"
              src="http://openweathermap.org/img/w/10n.png"
            />
            40
            °
            F
          </h3>
          <p>
            moderate rain
          </p>
        </div>,
      ],
      "className": "today-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Current weather in London, GB",
        },
        "ref": null,
        "rendered": "Current weather in London, GB",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              <img
                alt="moderate rain"
                src="http://openweathermap.org/img/w/10n.png"
              />
              40
              °
              F
            </h3>,
            <p>
              moderate rain
            </p>,
          ],
          "className": "today-weather",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="moderate rain"
                  src="http://openweathermap.org/img/w/10n.png"
                />,
                40,
                "°",
                "F",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "moderate rain",
                  "src": "http://openweathermap.org/img/w/10n.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              40,
              "°",
              "F",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "moderate rain",
            },
            "ref": null,
            "rendered": "moderate rain",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Current weather in London, GB
          </h2>,
          <div
            className="today-weather"
          >
            <h3>
              <img
                alt="moderate rain"
                src="http://openweathermap.org/img/w/10n.png"
              />
              40
              °
              F
            </h3>
            <p>
              moderate rain
            </p>
          </div>,
        ],
        "className": "today-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Current weather in London, GB",
          },
          "ref": null,
          "rendered": "Current weather in London, GB",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <img
                  alt="moderate rain"
                  src="http://openweathermap.org/img/w/10n.png"
                />
                40
                °
                F
              </h3>,
              <p>
                moderate rain
              </p>,
            ],
            "className": "today-weather",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="moderate rain"
                    src="http://openweathermap.org/img/w/10n.png"
                  />,
                  40,
                  "°",
                  "F",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "moderate rain",
                    "src": "http://openweathermap.org/img/w/10n.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                40,
                "°",
                "F",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "moderate rain",
              },
              "ref": null,
              "rendered": "moderate rain",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
